content-jumpbox-provisioner/cloudformation/stack.yml

```diff
@@ -0,0 +1,122 @@
+           Parameters:
+              - TagEnvironment
+              - TagTeamDL
+              - TagSystemCode
+              - TagIpCode
+              - TagDescription
+ 
+ Parameters:
+     SubnetIds:
+         Description: List of comma separated subnet IDs
+         Type: CommaDelimitedList
+
+     SecurityGroups:
+         Description: List of comma separated secudity group IDs
+         Type: CommaDelimitedList
+ 
+     EnvironmentName:
+         Description: An environment name that will be prefixed to resource names
+         Type: String
+ 
+     EnvironmentType:
+         Description: Tag detail for the Environment
+         Type: String
+         Default: p
+ 
+     Ec2InstanceType:
+         Description: Size of ec2 Instance
+         Type: String
+         Default: t2.micro
+
+     ImageId:
+         Description: AMI to be used for the jumpbox
+         Type: String
+         Default: ami-e4515e0e
+ 
+     TagTeamDL:
+         Description: Tag of the TeamDL
+         Type: String
+         AllowedPattern: ^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$
+         ConstraintDescription: There must be a valid email address for the TeamDL
+         Default: universal.publishing.platform@ft.com
+ 
+     TagSystemCode:
+         Description: The system code for the environment
+         Type: String
+         Default: upp
+ 
+     TagIpCode:
+         Description: The environment ipCode
+         Type: String
+         AllowedPattern: '[P][0-9]*'
+         Default: P196
+ 
+     TagDescription:
+         Description: Tag detail for the describing the instance
+         Type: String
+         Default: UPP Jumpbox for Ops
+ 
+ Resources:
+   AutoScalingGroup:
+     Type: AWS::AutoScaling::AutoScalingGroup
+     Properties:
+       LaunchConfigurationName:
+         Ref: LaunchConfig
+       MinSize: '1'
+       MaxSize: '1'
+       VPCZoneIdentifier: !Ref SubnetIds
+       Tags:
+           - Key: Name
+             Value: !Join ["" , ["Upp Jumpbox ", !Ref EnvironmentType]]
+             PropagateAtLaunch: true
+           - Key: environment
+             Value: !Ref EnvironmentType
+             PropagateAtLaunch: true
+           - Key: teamDL
+             Value: !Ref TagTeamDL
+             PropagateAtLaunch: true
+           - Key: systemCode
+             Value: !Ref TagSystemCode
+             PropagateAtLaunch: true
+           - Key: ipCode
+             Value: !Ref TagIpCode
+             PropagateAtLaunch: true
+           - Key: description
+             Value: !Ref TagDescription
+             PropagateAtLaunch: true
+   LaunchConfig:
+     Type: AWS::AutoScaling::LaunchConfiguration
+     Properties:
+       IamInstanceProfile: "FT-Linux-Role"
+       ImageId: !Ref ImageId
+       InstanceType: !Ref Ec2InstanceType
+       KeyName: "upp-k8s-provisioning-debug"
+       SecurityGroups: !Ref SecurityGroups
+       UserData:
+         Fn::Base64: !Sub |
+           #!/bin/bash -x
```

Comment:
We might need to add a tweak to the SSHD config to allow agent forwarding when SSHing to cluster instances.

See here for more details:
http://git.svc.ft.com/projects/CP/repos/coco_jumpoff/browse/manifests/ssh_config.pp
http://git.svc.ft.com/projects/CP/repos/coco_jumpoff/browse/files/ssh_config

If you've tested it and we don't need those changes, then that's cool 👍

Reply:
this is just a first pass of provisioning a jumpbox for ops to use. haven't gotten to implementing tunnel functionalities with this jumpbox.

Reply:
Cool. 👍